### PR TITLE
[Modular] Mob layer shifting verbs

### PR DESCRIPTION
**Pull request by NotRanged**
## About The Pull Request

Adds two verbs to the IC tab which allow you to slightly increment your mob's 'layer' var. Useful for situations where you want a certain pixelshifted character to be drawn above or below the others.

## Why It's Good For The Game

yes

## Changelog
:cl:
add: Added two verbs to the IC tab that change your mob's 'layer' (sprite background/foreground priority) slightly. 
/:cl:

### CODE REVIEW

**Comment by Gandalf2k15** [comment]
When the player moves can we reset the layer to 4?

File: modular_skyrat/modules/layer_shift/code/modules/mob/mob_movement.dm [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#define MOB_LAYER_SHIFT_INCREMENT	0.01
+#define MOB_LAYER_SHIFT_MIN 		3.95
+//#define MOB_LAYER 				4   // This is a byond standard define
+#define MOB_LAYER_SHIFT_MAX   		4.05
+
+/mob/verb/layershift_up()
+	set name = "Shift Layer Upwards"
+	set category = "IC"
+
+	if(incapacitated() || (layer == LYING_MOB_LAYER))
```
Comment (Azarak) [inline modular_skyrat/modules/layer_shift/code/modules/mob/mob_movement.dm]:
```suggestion
	if(incapacitated() || body_position == LYING_DOWN)
```

File: modular_skyrat/modules/layer_shift/code/modules/mob/mob_movement.dm [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+	layer += MOB_LAYER_SHIFT_INCREMENT
+	var/layer_priority = (layer - MOB_LAYER) * 100 // Just for text feedback
+	to_chat(src, "<span class='notice'>Your layer priority is now [layer_priority].</span>")
+
+/mob/verb/layershift_down()
+	set name = "Shift Layer Downwards"
+	set category = "IC"
+
+	if(incapacitated() || (layer == LYING_MOB_LAYER))
```
Comment (Azarak) [inline modular_skyrat/modules/layer_shift/code/modules/mob/mob_movement.dm]:
```suggestion
	if(incapacitated() || body_position == LYING_DOWN)
```

**Review by Azarak — COMMENTED** [review]
Might make the verbs go off mob/living/ instead for the body position vars to work, I dont remember of which did they inherit from

**Comment by ShadeAware** [comment]
I have some concerns. Would this allow players to change their layer below objects to hide beneath them? Say you were atop a table, and shifted your sprite layer down, becoming ingrained beneath the table and invisible.

**Comment by Azarak** [comment]
> 
> 
> When the player moves can we reset the layer to 4?

I dont think this nessecary, moving gets called often and with this you can barely get your order to change, it will only change with things that also share the MOB layer, since it's such a small window. The layers will be reset as people lay down or stand up either way.

**Comment by Azarak** [comment]
> 
> 
> I have some concerns. Would this allow players to change their layer below objects to hide beneath them? Say you were atop a table, and shifted your sprite layer down, becoming ingrained beneath the table and invisible.

It'll only matter for other things that share the `MOB_LAYER`

**Comment by Echopsychotic** [comment]
I love that this is now being added.

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**